src/cogent3/core/new_alignment.py

```diff
@@ -556,17 +586,21 @@ def to_dict(self, as_array: bool = False) -> dict[str, Union[str, numpy.ndarray]
         get = self.seqs.get_seq_array if as_array else self.seqs.get_seq_str
         return {name: get(seqid=name) for name in self.names}
 
-    def degap(self, **kwargs):
-        """Returns copy in which sequences have no gaps.
+    def degap(self) -> SequenceCollection:
+        """Returns new collection in which sequences have no gaps.
 
-        Parameters
-        ----------
-        kwargs
-            passed to class constructor
+        Notes
+        -----
+        The returned collection will not retain an annotation_db if present.
         """
         # refactor: array - chars > gap char
-        seqs = {name: self.seqs[name].degap() for name in self.names}
-        return make_unaligned_seqs(seqs, moltype=self.moltype, info=self.info, **kwargs)
+        seqs = {
+            name: self.moltype.degap(self.seqs.get_seq_array(seqid=name))
+            for name in self.names
+        }
+        return make_unaligned_seqs(
```

Comment:
add a comment to refactor. Within a class, methods that return self should use `self.__class__(...)`, using that function raises the prospect of circular imports.

Reply:
ah yes I was wondering about this